pkg/eosclient/eosgrpc/eoshttp.go

```diff
@@ -373,12 +373,20 @@ func (c *EOSHTTPClient) GETFile(ctx context.Context, remoteuser string, auth eos
 }
 
 // PUTFile does an entire PUT to upload a full file, taking the data from a stream.
-func (c *EOSHTTPClient) PUTFile(ctx context.Context, remoteuser string, auth eosclient.Authorization, urlpath string, stream io.ReadCloser, length int64, app string) error {
+func (c *EOSHTTPClient) PUTFile(ctx context.Context, remoteuser string, auth eosclient.Authorization, urlpath string, stream io.ReadCloser, length int64, app string, disableVersioning bool) error {
 	log := appctx.GetLogger(ctx)
 	log.Info().Str("func", "PUTFile").Str("remoteuser", remoteuser).Str("uid,gid", auth.Role.UID+","+auth.Role.GID).Str("path", urlpath).Int64("length", length).Str("app", app).Msg("")
 
 	// Now send the req and see what happens
-	finalurl, err := c.buildFullURL(urlpath, auth)
+	tempUrl, err := c.buildFullURL(urlpath, auth)
+	base, err := url.Parse(tempUrl)
+	queryValues := base.Query()
+
+	if disableVersioning {
+		queryValues.Add("eos.versioning", strconv.Itoa(0))
+	}
+	finalurl := base.String() + queryValues.Encode()
```

Comment:
If `base` contains query parameters, you are duplicating them.

You might want to `base.RawQuery = queryValues.Encode()` and then just `finalurl := base.String()`

Reply:
Also, you don't seem to treat the `err` that `buildFullURL` might give.